ui/src/app/views/breadcrumb/component.tsx

```diff
@@ -0,0 +1,133 @@
+import {Component} from '@angular/core';
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+
+import {ReactWrapperBase, withCurrentCohort, withCurrentConceptSet, withCurrentWorkspace, withRouteConfigData, withUrlParams} from 'app/utils/index';
+import {navigateByUrl} from 'app/utils/navigation';
+
+const styles = {
+  firstLink: {
+    color: '#2691D0',
+    textDecoration: 'none'
+  },
+  lastLink: {
+    color: '#262262',
+    fontWeight: 600,
+    fontSize: '1rem',
+    textDecoration: 'none'
+  }
+};
+
+export const getTrail = (type, data) => {
+  const {workspace, cohort, conceptSet, urlParams: {ns, wsid, cid, csid, pid, nbName}} = data;
+  const prefix = `/workspaces/${ns}/${wsid}`;
+  switch (type) {
+    case 'workspaces':
+      return [
+        {label: 'Workspaces', url: '/workspaces'}
+      ];
+    case 'workspace':
+      return [
+        ...getTrail('workspaces', data),
+        {label: workspace ? workspace.name : '...', url: prefix}
+      ];
+    case 'workspaceEdit':
+      return [
+        ...getTrail('workspace', data),
+        {label: 'Edit Workspace', url: `${prefix}/edit`}
+      ];
+    case 'workspaceClone':
+      return [
+        ...getTrail('workspace', data),
+        {label: 'Duplicate Workspace', url: `${prefix}/clone`}
+      ];
+    case 'notebook':
+      return [
+        ...getTrail('workspace', data),
+        {label: 'Notebooks', url: `${prefix}/notebooks`},
```

Comment:
There was some intentional logic based on Lou's mocks implemented by the `intermediate: true` approach we had previously. AFAICT that's lost here, though I'm not 100% sure from tracing the code.

Reply:
I was careful to capture this logic in the new version. For example, the `notebooks` page has the `Workspace` breadcrumb, while the `notebooks/foo` page has the `Notebook` breadcrumb, which contains both the former 'intermediate' crumb and the new leaf.